bebraven/platform
# Fix Add User functionality

**Task**: https://app.asana.com/0/1174274412967132/1198168803811457

### Description

Creating users from `platformweb/users/new` was broken. This PR fixes it. 

### Motivation and Context

Before this PR, `Add User` was generating the following Devise error:

```
Filter chain halted as :require_no_authentication rendered or redirected
```
When we're creating a user from `platformweb/users/new`, we shouldn't be going through the Devise route. We're really just offering a UI for what we do via Rails console. 

But because we moved the `post users#create` route out of the `admin` namespace in #350, Devise is redirecting us to `Users::RegistrationsController#create`. 

For more details on the Devise error (which wasn't necessary to fix since we re-routed): based on [this bug report](https://github.com/heartcombo/devise/issues/4524), Devise's default behavior is to prevent creating new users (e.g., user registration) when you are already signed in. It does this [here](https://github.com/heartcombo/devise/blob/e3a00b27d19ba995891d7dd92394fe2900a789c2/app/controllers/devise/registrations_controller.rb#L4), in the `Devise::RegistrationsController` (which we inherit from in `Devise::RegistrationsController`).

### Test

- Sign in to `platformweb`, click **Users**, then **Add User**
- Fill out form, click **Add User**
<img width="1792" alt="Screen Shot 2020-10-13 at 3 02 34 PM" src="https://user-images.githubusercontent.com/62911934/95921117-8fc76700-0d65-11eb-822d-1aa021de9173.png">

- See that you're redirected to `platformweb/users` and see that the new user is in the list
<img width="1792" alt="Screen Shot 2020-10-15 at 12 00 48 PM" src="https://user-images.githubusercontent.com/62911934/96174453-2d45a680-0ede-11eb-962f-68108b05d8a6.png">

### Types of changes




### Checklist:
